script.js

```diff
@@ -1,4 +1,273 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+// ❌⚔🤺
+
+const DEFAULT_SETTINGS = {
+  highScoreMode: false,
+  knockoutMode: false,
+  numPlayers: 10,
+  numDice: 2,
+  round: 1,
+  players: [],      // used for knockout mode, players not yet eliminated
+  pointerX: -1,     // used for knockout mode, index of playerX
+  pointerY: -1,     // used for knockout mode, index of playerY
+  playerRolls: [],  // knockout mode - stores rolls and scores of a round
+                    // regular mode - stores scores of a round
+  playerScores: [], // used for regular mode, cumulative scores of players
+}
+
+var S = {...DEFAULT_SETTINGS}
+
+// HELPER FUNCTIONS
+var diceRoll = function () {
+  var roll = []
+  
+  for (let i = 0; i < S.numDice; i += 1) {
+    roll.push(Math.floor(Math.random() * 6) + 1)
+  }
+  // return an array containing dice rolls
+  return roll
+}
+
+var rollMax = function(arr, highScoreMode) {
+  // returns score based on high score mode
+  if (highScoreMode) return Number([...arr].sort().reverse().join(""))
+  else return Number([...arr].sort().join(""))
+}
+
+var selectKnockoutPlayers = function() {
+  
+  do {
+  S.pointerX = Math.floor(Math.random() * S.players.length)
+  S.pointerY = Math.floor(Math.random() * S.players.length)
+  } while (S.pointerX === S.pointerY)
+  
+  return `Round ${S.round}<br><br>
+  Player ${S.players[S.pointerX] + 1} VS Player ${S.players[S.pointerY] + 1}<br><br>
+  Player ${S.players[S.pointerX] + 1}, roll the dice.`
+
+}
```

Comment:
looks v familiar but hope u understand how this works and what the `do/while` loops does + how its different from a `while` loop